spec/unit/info_service_spec.rb

```diff
@@ -33,6 +33,26 @@
         end
       end
 
+      it "returns task data for valid tasks in an environment even if invalid tasks exist" do
+        Puppet.override(:environments => env_loader) do
+          @mod = PuppetSpec::Modules.create(mod_name, modpath, {:environment => env,
+                                                                :tasks => [['atask',
+                                                                            {:name => 'atask.json',
+                                                                             :content => metadata.to_json}],
+                                                                           ['btask',
+                                                                            {:name => 'btask.json',
+                                                                             :content => metadata.to_json}],
+                                                                           ['ctask',
+                                                                            {:name => 'ctask.json',
+                                                                             :content => metadata.to_json}]]})
+          File.write("#{modpath}/#{mod_name}/tasks/atask.json", "NOT JSON")
+          expect(Puppet).to receive(:send_log).with(:err, 'Failed to validate task')
+          @tasks = Puppet::InfoService.tasks_per_environment(env_name)
+
+          expect(@tasks.map{|t| t[:name]}.sort).to eq(%w{test1::btask test1::ctask})
```

Comment:
not blocking, but `expect([1, 2]).to contain_exactly(2, 1)` can be used to match arrays ignoring order